File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_VBNet/objetosgui/Dialogo.vb
'
' * Created by SharpDevelop.
' * User: henrique.souza
' * Date: 07/07/2015
' * Time: 14:01
' * 
' * To change this template use Tools | Options | Coding | Edit Standard Headers.
' 

Imports System.Drawing
Imports System.Windows.Forms

Namespace objetosgui
	''' <summary>
	''' Description of Dialogo.
	''' </summary>
	Public Class Dialogo
		Public Sub New()
		End Sub

		Public Shared Sub mensagemErro(texto As [String])
			MessageBox.Show(texto, "Erro", MessageBoxButtons.OK, MessageBoxIcon.[Error])
		End Sub

		Public Shared Sub mensagemInfo(texto As [String])
			MessageBox.Show(texto, "Informação", MessageBoxButtons.OK, MessageBoxIcon.Information)
		End Sub

		Public Shared Function confirma(texto As [String]) As Boolean
			Dim res As DialogResult = MessageBox.Show(texto, "Confirma", MessageBoxButtons.YesNo, MessageBoxIcon.Question)
			Return (res = DialogResult.Yes)
		End Function

		Public Shared Function entrada(texto As [String], valorInicial As [String]) As [String]
			Dim form As New Form()
			Dim label As New Label()
			Dim textBox As New TextBox()
			Dim buttonOk As New Button()
			Dim buttonCancel As New Button()

			form.Text = "Informa"
			label.Text = texto
			textBox.Text = valorInicial
			textBox.MaxLength = 10

			buttonOk.Text = "OK"
			buttonCancel.Text = "Cancel"
			buttonOk.DialogResult = DialogResult.OK
			buttonCancel.DialogResult = DialogResult.Cancel

			label.SetBounds(9, 20, 372, 13)
			textBox.SetBounds(12, 36, 372, 20)
			buttonOk.SetBounds(228, 72, 75, 23)
			buttonCancel.SetBounds(309, 72, 75, 23)

			label.AutoSize = True
			textBox.Anchor = textBox.Anchor Or AnchorStyles.Right
			buttonOk.Anchor = AnchorStyles.Bottom Or AnchorStyles.Right
			buttonCancel.Anchor = AnchorStyles.Bottom Or AnchorStyles.Right

			form.ClientSize = New Size(396, 107)
			form.Controls.AddRange(New Control() {label, textBox, buttonOk, buttonCancel})
			form.ClientSize = New Size(Math.Max(300, label.Right + 10), form.ClientSize.Height)
			form.FormBorderStyle = FormBorderStyle.FixedDialog
			form.StartPosition = FormStartPosition.CenterScreen
			form.MinimizeBox = False
			form.MaximizeBox = False
			form.AcceptButton = buttonOk
			form.CancelButton = buttonCancel

			Dim res As DialogResult = form.ShowDialog()

			If res = DialogResult.OK Then
				Return textBox.Text
			Else
				Return ""
			End If
		End Function

		Public Shared Function entrada(texto As [String]) As [String]
			Return entrada(texto, "")
		End Function

	End Class
End Namespace

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_CSharp/objetosgui/Dialogo.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 07/07/2015
 * Time: 14:01
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using System.Drawing;
using System.Windows.Forms;

namespace HFSGuardaDiretorio.objetosgui
{
	/// <summary>
	/// Description of Dialogo.
	/// </summary>
	public class Dialogo
	{
		public Dialogo()
		{
		}
		
		public static void mensagemErro(String texto) {
			MessageBox.Show(texto, "Erro", MessageBoxButtons.OK, MessageBoxIcon.Error);
		}
		
		public static void mensagemInfo(String texto) {
			MessageBox.Show(texto, "Informação", MessageBoxButtons.OK, MessageBoxIcon.Information);
		}
		
		public static bool confirma(String texto) {
			DialogResult res = MessageBox.Show(texto, "Confirma", MessageBoxButtons.YesNo, MessageBoxIcon.Question); 			
			return (res == DialogResult.Yes);			 
		}
		
		public static String entrada(String texto, String valorInicial) {
			Form form = new Form();
			Label label = new Label();
			TextBox textBox = new TextBox();
			Button buttonOk = new Button();
			Button buttonCancel = new Button();
			
			form.Text = "Informa";
			label.Text = texto;
			textBox.Text = valorInicial;
			textBox.MaxLength = 10;
			
			buttonOk.Text = "OK";
			buttonCancel.Text = "Cancel";
			buttonOk.DialogResult = DialogResult.OK;
			buttonCancel.DialogResult = DialogResult.Cancel;
			
			label.SetBounds(9, 20, 372, 13);
			textBox.SetBounds(12, 36, 372, 20);
			buttonOk.SetBounds(228, 72, 75, 23);
			buttonCancel.SetBounds(309, 72, 75, 23);
			
			label.AutoSize = true;
			textBox.Anchor = textBox.Anchor | AnchorStyles.Right;
			buttonOk.Anchor = AnchorStyles.Bottom | AnchorStyles.Right;
			buttonCancel.Anchor = AnchorStyles.Bottom | AnchorStyles.Right;
			
			form.ClientSize = new Size(396, 107);
			form.Controls.AddRange(new Control[] { label, textBox, buttonOk, buttonCancel });
			form.ClientSize = new Size(Math.Max(300, label.Right + 10), form.ClientSize.Height);
			form.FormBorderStyle = FormBorderStyle.FixedDialog;
			form.StartPosition = FormStartPosition.CenterScreen;
			form.MinimizeBox = false;
			form.MaximizeBox = false;
			form.AcceptButton = buttonOk;
			form.CancelButton = buttonCancel;
			
			DialogResult res = form.ShowDialog();
			
			if (res == DialogResult.OK)
				return textBox.Text;
			else
				return "";
		}
		
		public static String entrada(String texto) {
			return entrada(texto, "");
		}
		
	}
}
